src/transform.js

```diff
@@ -646,84 +645,50 @@ function untransformObject(schema, className, mongoObject, isNestedObject = fals
 
     var restObject = untransformACL(mongoObject);
     for (var key in mongoObject) {
-      // TODO: This could use some cleanup. Default case handling still needs to happen for some nested keys.
+      const specialKeys = [
+        '_id',
+        '_hashed_password',
+        '_acl',
+        '_email_verify_token',
+        '_perishable_token',
+        '_tombstone',
+        '_session_token',
+        'updatedAt',
+        '_updated_at',
+        'createdAt',
+        '_created_at',
+        'expiresAt',
+        '_expiresAt',
+      ];
+      if (_.includes(specialKeys, key) && isNestedObject) {
```

Comment:
should switch these two statements around so the if will exit early with the boolean compare rather than a Object key iteration

```javascript
if(isNestedObject && _.includes(specialKeys, key)) {
...
}
``